### ISSUE TITLE
Tests fail on node v19

### ISSUE THREAD

**Issue by hildjj** [hildjj/mock-tls-server#2]
Symptom:

```
  Uncaught exception in test/index.ava.js

  TypeError: Cannot read properties of null (reading 'finishShutdown')
```

this is from https://github.com/nodejs/node/blob/d579bc16b3c7e587d6027f8ea7a187781d448e4f/lib/internal/js_stream_socket.js#L160:

```js
 finishShutdown(handle, errCode) {
    // The shutdown request might already have been cancelled.
    if (this[kCurrentShutdownRequest] === null)
      return;
    const req = this[kCurrentShutdownRequest];
    this[kCurrentShutdownRequest] = null;
    handle.finishShutdown(req, errCode); // `handle` is null
  }
```

I think this means that we're shutting down the socket twice, but I'd argue that node should catch that and deal with it by firing an error or something, rather than crashing.

I've tried removing https://github.com/hildjj/mock-tls-server/blob/fde1f8c1f837e1fc7886232a4cdf253a8a19256f/lib/socket.js#L108, which fixes this issue, but breaks other tests.

My current approach is to try to understand the actual flow well enough to create a small repro for the node team, then file a bug there.

**Comment by hildjj** [comment]
Might be related to https://github.com/nodejs/node/issues/35695

**Comment by hildjj** [comment]
Fixed in #3
